sorting/wiggle_sort.cpp

```diff
@@ -0,0 +1,136 @@
+/**
+ * \addtogroup sorting Sorting Algorithms
+ * @{
+ * \file
+ * \brief [Wiggle Sort Algorithm]
+ * (https://leetcode.com/problems/wiggle-sort-ii/) Implementation
+ *
+ * \author [Roshan Kanwar](http://github.com/roshan0708)
+ *
+ * \details
+ * Wiggle Sort sorts the array into a wave like array.
+ * An array ‘arr[0..n-1]’ is sorted in wave form,
+ * if arr[0] >= arr[1] <= arr[2] >= arr[3] <= arr[4] >= …..
+ *
+ * \example
+ * arr = [1,1,5,6,1,4], after wiggle sort arr will become equal to [1,1,6,1,5,4]
+ * arr = [2,8,9,1,7], after wiggle sort arr will become equal to [8,2,9,1,7]
+ */
+
+#include <iostream>  /// for io operations
+#include <algorithm>
+#include <vector>
+#include <cassert>
+#include <ctime>
+
+/**
+ * @namespace sorting
+ * @brief Sorting algorithms
+ */
+namespace sorting {
+    /**
+    * @namespace wiggle_sort
+    * @brief Functions for [Wiggle Sort](https://leetcode.com/problems/wiggle-sort-ii/) algorithm
+    */
+    namespace wiggle_sort {
+
+        /**
+        *
+        * @brief Function used for sorting the elements in wave form.
+        * @details
+        * Checking whether the even indexed elements are greater than
+        * their adjacent odd elements.
+        * Traversing all even indexed elements of the input arr.
+        * If current element is smaller than the previous odd element, swap them.
+        * If current element is smaller than the next odd element, swap them.
+        *
+        * @param arr input array (unsorted elements)
+        *
+        */
+        template<typename T> // this allows to have vectors of ints, double, float, etc
+        std::vector<T> wiggleSort(const std::vector<T> &arr) {
```

Comment:
> help with the error

The fix below should remove the warnings.
```suggestion
        std::vector<T> wiggleSort(std::vector<T> arr) {
```